cypress/integration/interpretations.cy.js

```diff
@@ -0,0 +1,169 @@
+
+const TEST_AO_ID = TEST_AOS[0].id
+
+const TEST_CANCEL_LABEL = 'Cancel'
+const TEST_POST_INTERPRETATION_LABEL = 'Post interpretation'
+const TEST_WRITE_INTERPRETATION_LABEL = 'Write an interpretation'
+const TEST_WRITE_REPLY_LABEL = 'Write a reply'
+const TEST_INTERPRETATION_TEXT = 'Test interpretation'
+const TEST_INTERPRETATION_TEXT_EDITED = `${TEST_INTERPRETATION_TEXT} (edited)`
+const TEST_INTERPRETATION_COMMENT_TEXT = 'Reply to test interpretation'
+
+describe('interpretations', () => {
+    it('the interpretations panel can be toggled when clicking the button in the toolbar', () => {
+        cy.visit(`#/${TEST_AO_ID}`, EXTENDED_TIMEOUT)
+
+        expectInterpretationsButtonToBeEnabled()
+
+        clickMenubarInterpretationsButton()
+
+        cy.getBySel('details-panel').should('be.visible')
+
+        clickMenubarInterpretationsButton()
+
+        cy.getBySel('details-panel').should('not.exist')
+
+        clickMenubarInterpretationsButton()
+
+        // Interpretations section is visible
+        cy.getBySel('details-panel').should('contain', 'Interpretations')
+
+        // default form for adding interpretation is visible
+        expectInterpretationFormToBeVisible()
+    })
+
+    it('a new interpretation can be added', () => {
+        // the rich text editor shows when clicking the input
+        cy.getBySel('interpretation-form')
+            .find(`input[placeholder="${TEST_WRITE_INTERPRETATION_LABEL}"]`)
+            .click()
+
+        cy.getBySel('interpretation-form').should(
+            'contain',
+            TEST_POST_INTERPRETATION_LABEL
+        )
+        cy.getBySel('interpretation-form').should('contain', TEST_CANCEL_LABEL)
+
+        // the rich text editor is removed when clicking Cancel
+        cy.getBySel('interpretation-form').contains(TEST_CANCEL_LABEL).click()
+
+        cy.getBySel('interpretation-form').should(
+            'not.contain',
+            TEST_POST_INTERPRETATION_LABEL
+        )
+        cy.getBySel('interpretation-form').should(
+            'not.contain',
+            TEST_CANCEL_LABEL
+        )
+
+        // it's possible to write a new interpretation text
+        cy.getBySel('interpretation-form')
+            .find(`input[placeholder="${TEST_WRITE_INTERPRETATION_LABEL}"]`)
+            .click()
+
+        typeTextarea('interpretation-form', TEST_INTERPRETATION_TEXT)
+
+        // the new interpretation can be saved and shows up in the list
+        cy.getBySel('interpretation-form')
+            .contains(TEST_POST_INTERPRETATION_LABEL)
+            .click()
+
+        cy.getBySel('interpretations-list').should(
+            'contain',
+            TEST_INTERPRETATION_TEXT
+        )
+        cy.getBySel('interpretations-list').should(
+            'contain',
+            'See interpretation'
+        )
+    })
+
+    it('the new interpretation can be edited', () => {
+        cy.getBySel('interpretation-edit-button').first().click()
```

Comment:
why `first`?

Reply:
Yes, but since this is an existing AO that we open, are we always 100% sure there isn't any other interpretation?

Reply:
if you pick out an existing AO that only has 1 interpretation then there should only be 1 unless we explicitly change this in the db template, which we shouldn't. So I'd say it's actually a good thing if it fails because there are more than 1, since that means that something is obviously wrong.

Reply:
the reason for using https://test.e2e.dhis2.org/lytics-apps-dev/ is that it should only be used by automated tests, which in turn should be self-cleaning. So the db can be expected to be exactly the same over time.

Reply:
Removed `first()` 👍